src/features/courses/course-route.ts

```diff
@@ -642,6 +643,29 @@ router.get('/question/:id/grade',
 
     }));
 
+router.get('/question/:id/openlab',
+    authenticationMiddleware,
+    validate(getQuestionOpenLabValidation),
+    // This is a typescript workaround since it tries to use the type extractMap
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    asyncHandler(async (req: RederlyExpressRequest<any, unknown, GetQuestionGradeRequest.body, unknown>, res: Response, next: NextFunction) => {
+        if (_.isNil(req.session)) {
+            throw new RederlyError(Constants.ErrorMessage.NIL_SESSION_MESSAGE);
+        }
+
+        const user = await req.session.getUser();
+        const { id: questionId } = req.params as GetQuestionRequest.params;
+        const baseURL = req.headers['rederly-origin'] as string | undefined; // need this because it incorrectly thinks it can be an array
```

Comment:
We should probably use an enum or string constants instead of hardcoded strings.